riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/DefaultRiptideRegistrar.java

```diff
@@ -394,15 +395,19 @@ private Tag clientName(final String id, final Client client) {
     }
 
     private BeanMetadataElement trace(final String executor) {
-        if (registry.isRegistered("tracer")) {
-            return genericBeanDefinition(TracingExecutors.class)
-                    .setFactoryMethod("preserve")
-                    .addConstructorArgReference(executor)
-                    .addConstructorArgReference("tracer")
-                    .getBeanDefinition();
-        } else {
-            return ref(executor);
-        }
+        final AtomicReference<BeanMetadataElement> result = new AtomicReference<>();
```

Comment:
Ah, that's unfortunate. Can you implement a new version of `ifPresent`?

Signatures that I could imagine that would work:

- `<T> T ifPresent(String, Supplier<T>, Supplier<T>)`
- `<T> Optional<T>(String, Supplier<T>)`

That would remove the need for a stateful reference.